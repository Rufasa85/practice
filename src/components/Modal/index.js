import React from 'react'

export default function Modal(props) {
    return (
        <div className="modalBackdrop">
            <div className="modalContainer">
                <h3 className="modalTitle">{props.photo.name}</h3>
                <img alt={props.photo.category}src={props.photo.src} />
                <p>
                    {props.photo.desc}
    </p>
                <button data-testid="closeme" onClick={props.close}type="button">
                    Close this modal
    </button>
            </div>
        </div>
    )
}
