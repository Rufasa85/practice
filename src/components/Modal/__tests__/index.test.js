import React from 'react'
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Modal from '..';

const picture =  {
    name: 'Grocery aisle',
    category: 'commercial',
    desc: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie',
 src:"null"
  };

afterEach(cleanup)

describe('Modal is rendering', () => {

  it('renders', () => {
    render(<Modal photo={picture} />);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Modal photo={picture} />);
    expect(asFragment()).toMatchSnapshot()
  })
  it('renders', () => {
    const { getByTestId } = render(<Modal photo={picture} />)
    expect(getByTestId('closeme')).toHaveTextContent('Close this modal')
  })
})