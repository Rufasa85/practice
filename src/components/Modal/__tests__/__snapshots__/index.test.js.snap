// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal is rendering matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="modalTitle"
      >
        Grocery aisle
      </h3>
      <img
        alt="commercial"
        src="null"
      />
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie
      </p>
      <button
        data-testid="closeme"
        type="button"
      >
        Close this modal
      </button>
    </div>
  </div>
</DocumentFragment>
`;
